src/Microsoft.ApplicationInsights.AspNet/Extensions/HttpRequestExtensions.cs

```diff
@@ -0,0 +1,53 @@
+﻿namespace Microsoft.ApplicationInsights.AspNet.Extensions
+{
+    using Microsoft.AspNet.Http;
+    using System;
+    using System.Text;
+
+    /// <summary>
+    /// Set of extension methods for Microsoft.AspNet.Http.HttpRequest
+    /// </summary>
+    public static class HttpRequestExtensions
```

Comment:
`HttpRequestExtensions` is used only by the `RequestTrackingMiddleware` correct? Why do we want it to be public? 
This imposes the requirement to test arguments and throw exceptions here, leading to ask by @SergKanz to catch them in the `RequestTrackingMiddleware`.